worker/cmd/generate.go

```diff
@@ -175,10 +173,15 @@ var generateCmd = &cobra.Command{
 						sugar.Errorf("Could not pop from redis queue: %v", err)
 						continue
 					}
-					jobChan <- job
+					NewJobProcessor(ctx, pool, svc, sugar, job,
+						PreCheckEndpointURL,
+						SdgEndpointURL,
+						TlsClientCertPath,
+						TlsClientKeyPath,
+						TlsServerCaCertPath).processJob()
```

Comment:
continuing suggested changes from above ...

```suggestion
					jobInProgress = true
					jobChan <- job
```